File: src/java/org/slf4j/impl/StaticMDCBinder.java
package org.slf4j.impl;

import org.slf4j.helpers.NOPMDCAdapter;
import org.slf4j.spi.MDCAdapter;

/**
 * This implementation is bound to {@link NOPMDCAdapter}.
 *
 */
public class StaticMDCBinder {

    /**
     * The unique instance of this class.
     */
    public static final StaticMDCBinder SINGLETON = new StaticMDCBinder();

    private StaticMDCBinder() {
    }

    /**
     * Currently this method always returns an instance of 
     * {@link NOPMDCAdapter}.
     *
     * @return instance of NOPMDCAdapter
     */
    public MDCAdapter getMDCA() {
        return new NOPMDCAdapter();
    }

    public String getMDCAdapterClassStr() {
        return NOPMDCAdapter.class.getName();
    }
}

File: src/slf4j_timbre/static_mdc_binder.clj
(ns slf4j-timbre.static-mdc-binder
	(:gen-class
		:name org.slf4j.impl.StaticMDCBinder
		:factory getSingleton
		:methods
			[[getMDCA [] org.slf4j.spi.MDCAdapter]
			 [getMDCAdapterClassStr [] String]])
	(:import org.slf4j.helpers.NOPMDCAdapter))

(defn -getMDCA
	[_]
	(NOPMDCAdapter.))

(defn -getMDCAdapterClassStr
	[_]
	(.getName NOPMDCAdapter))